### PR TITLE
Use and configure matplotlib sphinx extension

### PR DESCRIPTION
**Pull request by jl-wynen**
This can be used by putting this in a docstring:
```
    Examples
    --------
    
    .. plot::

      >>> sc.plot(sc.array(dims=['x'], values=[1, 2, 1]))
```

From a private slack discussion:
> We could write a scraper on top that automatically puts `.. plot::` if it detects `\.plot\(` or a similar pattern. numpydoc does this based on detecting import matplotlib.

Note that importing numpy in `plot_pre_code` is not strictly necessary. I added it in case it is needed by an example.

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
This is not working for me, I get `Failed example ...` from Sphinx (complaining that it got a plot object). Do I have to install something?

**Comment by SimonHeybrock** [comment]
I cannot get plots to show up consistently. I have tried many different options of where to place the directive and the code relative to each other.

**Comment by jl-wynen** [comment]
Found a way to use variables defined in a previous code block. See docs of `gaussian_filter`.

**Comment by SimonHeybrock** [comment]
Nice! I fixed some minor issues. In particular we have to use `close-figs` everywhere, otherwise you get the last plot from the previous page in addition to the plot from the current page.

**Comment by jl-wynen** [comment]
I don't like having to use a combination of lines starting with `>>>` and lines without. I have an idea for how do skip running doctest in those cases...

File: src/scipp/sphinxext/__init__.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            # if the block is already indented. That should not cause problems.
+            return [
+                ' ' * indentation_of(prev) + '.. plot::', '',
+                *map(lambda l: '    ' + l, block)
+            ]
+    return block
+
+
+def is_start_of_block(line: str) -> bool:
+    return line.lstrip().startswith('>>>')
```
Comment (SimonHeybrock) [inline src/scipp/sphinxext/__init__.py:36]: I suppose this breaks (something?) when not using doctest syntax?

Comment (jl-wynen) [reply src/scipp/sphinxext/__init__.py:36]: It only does anything when the code uses `>>>`. I have an alternative idea using the AST directly. But it does not look like matplotlib has proper support for that.

**Comment by jl-wynen** [comment]
Should pass doctest now. This seems like a dirty solution but it is very simple and does not require us rummaging around in matplotlib / doctest internals.

File: src/scipp/interpolate/__init__.py [hunk 2]
```diff
@@ -50,7 +50,9 @@ def _drop_masked(da, dim):
 def interp1d(da: DataArray,
              dim: str,
              *,
-             kind: Union[str, int] = 'linear',
+             kind: Union[int,
+                         Literal['linear', 'nearest', 'nearest-up', 'zero', 'slinear',
+                                 'quadratic', 'cubic', 'previous', 'next']] = 'linear',
```
Comment (SimonHeybrock) [inline src/scipp/interpolate/__init__.py:55]: Nice, I didn't know this is possible!

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**